fix(Verovio): Add missing methods

bbloomf opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://book.verovio.org/toolkit-reference/input-formats.html
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

mparisot-antescofo commented:
Thanks for the update

In the `redoLayout` funciton the `option: RedoLayoutOptions` was optional before, and the `resetCache` was optional too

Are you generating those definitions automatically or managing them manually? Because if it's the later I can also help by commiting on your fork to help you if needed.

All the other things I use were ok

bbloomf commented:
I updated them manually based on the documentation [here](https://book.verovio.org/toolkit-reference/toolkit-methods.html), as well as occasionally [verovio-toolkit.js](https://github.com/rism-digital/verovio/blob/develop/emscripten/npm/src/verovio-toolkit.js) from the verovio repository.

You're right though that in `redoLayout` the argument should be optional; I just updated this branch to reflect that.  I was also thinking that it might make more sense to submit a pull request to verovio itself to include the types directly in their NPM package.

mparisot-antescofo commented:
yeah that would be awesome to have it in the package directly, I'm not sure if they want to manage that or not as I think they compile things from C

Anyway all is ok on my side 👍

bbloomf commented:
Ready to merge from my perspective

bbloomf commented:
Ready to merge
